File: classes/parte-5-funciones-y-modularidad/082-alcance-scope-y-sombreado-shadowing/implementaciones/c/main.c
#include <stdio.h>

int main(void) {
    long n;
    if (scanf("%ld", &n) != 1) return 1;
    long x = n;
    {
        long x = n + 10; /* sombrea a la externa dentro del bloque */
        printf("interno=%ld externo=%ld\n", x, n);
    }
    return 0;
}

File: classes/parte-5-funciones-y-modularidad/082-alcance-scope-y-sombreado-shadowing/implementaciones/javascript/main.mjs
import { readFileSync } from "node:fs";

const n = parseInt(readFileSync(0, "utf8").trim(), 10);
const x = n;
{
  const x = n + 10; // sombrea a la externa dentro del bloque
  console.log(`interno=${x} externo=${n}`);
}
